### DISCUSSION TITLE
GreptimeDB commandline utilities

### DISCUSSION THREAD

**Discussion by v0y4g3r** [GreptimeTeam/greptimedb#3562] [Ideas]
Like MySQL's `mysqldump`/`mysqlcheck`, we need to provide a set of executable utlities for functionalities like:
- WAL dump & repair @v0y4g3r 
- Read manifest file of tables
- Backup database
- Import data in bulk load manner
- ...

**Comment by waynexia** [comment]
>Import data in bulk load manner

This should be a normal function of DB I think

**Comment by sunng87** [comment]
Consider implement it as subcommand of [gtctl](https://github.com/GreptimeTeam/gtctl).

@zyy17 I think we do need a subcommand system in gtctl to support various domains of task:

- kubernetes management
- db management
- cloud management

**Comment by zyy17** [comment]
@sunng87 What you mention above is also the design goal of gtctl. The DB can provide the fundamental mechanism that @v0y4g3r mentioned above and gtctl can integrate it in a more operator-friendly way. For convenience, the related binary utilities also can implement in greptimedb(maybe can share the same crate implementation), but such utilities are only for database developers.

**Comment by tisonkun** [comment]
Move to Ideas until we have more concrete proposals.
